portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java

```diff
@@ -69,14 +80,23 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
     /** Use query to populate a list of Comment objects. */
     ArrayList<Comment> comments = new ArrayList<Comment>();
-    
+
+    /** Comments may require translation. For now do it simply, improve later. */
+    Translate translate = TranslateOptions.getDefaultInstance().getService();
+
+    /** Add translated queries for response */
     for (Entity entity : commentResults.asIterable(options)) {
         long id = entity.getKey().getId();
         String message = (String) entity.getProperty("message");
         long timestamp = (long) entity.getProperty("timestamp");
         String email = (String) entity.getProperty("email");
 
-        Comment comment = new Comment(id, message, projectId, timestamp, email);
+        Translation translation = translate.translate(
+            message,
+            Translate.TranslateOption.targetLanguage(languageCode)
+        );
```

Comment:
Is this making a synchronous network request?  I assume that it is, in which case you don't want to be doing it within a loop on a potentially large number of entries.  I assume that there's either an ability to batch the translation requests together or else firing off async translation requests.

Reply:
Makes sense, it was added that way as MVP but I'll take a closer look at the API.

Reply:
After taking a closer look at this, looks like there's two options for doing it async, batch or threading.
- Batch
 1. It requires that source text is uploaded in one file to a cloud storage bucket, which isn't in itself a big deal, but would require an upload step, unclear how long that would take.
 2. It assumes all input is in the same language. This isn't strictly enforced on post...but other input languages might be enough of an edge case that it could be ignored.
- Threading
 1. Unclear whether translate is threadsafe, but could try and find out.

Reply:
Modified translation to be multithreaded.